### PR TITLE
[Build] Make mage-linux-amd64 statically compiled.

### PR DESCRIPTION
**Pull request by ph**
When I've upgraded my arch system, they have upgraded the libc library,
that libary is much newer than the library used in the crossbuild
docker images. This made building beats impossible because the
mage-linux-amd64 is compiled dynamically and used in all our docker
build.

This PR make the mage binary to be statically compiled so it doesn't
rely on any installed libraries.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## How to test this PR locally

Run mage package in any beats.

## Related issues

-

### CODE REVIEW

**Comment by ph** [comment]
Need to be backported to 7.x, 7.9.

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -20,6 +20,7 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
 - Ensure dynamic template names are unique for the same field. {pull}18849[18849]
 - Remove the deprecated `xpack.monitoring.*` settings. Going forward only `monitoring.*` settings may be used. {issue}9424[9424] {pull}18608[18608]
 - Added `certificate` TLS verification mode to ignore server name mismatch. {issue}12283[12283] {pull}20293[20293]
+- Make the mage binary used by the build process in the docker container to be statically compiled. {pull}20827[20827]
```
Comment (andrewkroh) [inline CHANGELOG.next.asciidoc:23]: Maybe this belongs in the developer changelog.

Comment (ph) [reply CHANGELOG.next.asciidoc:23]: good point, changing that.

**Comment by ph** [comment]
The failures in the macOS X seems valid, but not related to this changes created a follow issue. 
https://github.com/elastic/beats/issues/20836

**Pull request merged**